src/lib/firewall.ts

```diff
@@ -0,0 +1,198 @@
+import * as _ from 'lodash';
+
+import * as config from '../config/index';
+import * as constants from './constants';
+import * as iptables from './iptables';
+import { log } from './supervisor-console';
+
+import * as dbFormat from '../device-state/db-format';
+
+export const initialised = (async () => {
+	await config.initialized;
+	await applyFirewall();
+
+	// apply firewall whenever relevant config changes occur...
+	config.on('change', async ({ firewallMode, localMode }) => {
+		if (firewallMode || localMode !== undefined) {
```

Comment:
```suggestion
		if (firewallMode || localMode != null) {
```
I always encourage this until everything in TS in the supervisor (and even then I'd probably continue to).